### ISSUE TITLE
There was an error in XML (45, 6)

### ISSUE THREAD

**Issue by TonyGerace** [Azure/service-fabric-issues#276]
We have a somewhat larger solution (160+ projects, maybe 40 services). The build works however the deploy randomly fails with error
"There is an error in XML document (45, 6)."

The error seems random from developer to developer, cluster type (Local 1 Node vs Local 5 Node), and time.

While I believe the error to be something in one of the manifest - we cannot seem to pinpoint what is going on. Any help on this issue would be greatly appreciated.

Current development setup is
VS2017 - 15.2 (26430.6)
Service Fabric Tools 1.5.50311.1

**Comment by Mardoxx** [comment]
_Somewhat_, indeed!

I had this error come up a couple of times but I can not remember why it was... What does line 45 of your ServiceManifest.xml look like? (I assume this error is from your ServiceManifest)

**Comment by MikkelHegn** [comment]
@TonyGerace Can you share the output log from a failed deployment? Is this against a local dev cluster or any cluster?

**Comment by masnider** [comment]
@TonyGerace It would also be good if you could share the output of Test-ServiceFabricApplication -Verbose when this both succeeds and fails. If you could provide the version of the local SDK and remote Runtime of the cluster that you're targeting, that would be good data.

**Comment by TheLever** [comment]
I have the same issue cropping up today. The error seems to be that the schema of ApplicationManifest.xml is not respected. The problem is that it seems to be VS (2017) which keeps overwriting it according to some old schema - specifically, it seems to be the DefaultServices node it finds offending.

Service Fabric SDK - 2.6.210.494
Service Fabric - 5.6.210.9494

This debilitating issue is about a month old. I haven't been able to find a workaround or solution - can I provide any data to help diagnose and fix this one?

**Comment by MikkelHegn** [comment]
@TheLever Please provide the output of `Test-ServiceFabricApplication -Verbose` against the application package. If this is happening during build - please share a verbose build output log. Thanks

**Comment by cmpcdaly** [comment]
Hi Tony, I'm currently attempting to deploy an application with 34 services to a live SF cluster and running into various problems (updating seemed to stick on the first upgrade domain, and then deleting the application timed out and is currently stuck).

Just wondering if you could tell me how the application deployed with 40 services is getting on? Are we pushing the limits of a single service fabric application here or is there something else going on?

**Comment by oanapl** [comment]
@cmpcdaly , are the issues for the 34 services tracked with https://github.com/Azure/service-fabric-issues/issues/614? If so, let's continue the discussion there.
Service Fabric can handle applications with 34 services. We have seen way bigger applications. 
Of course, it matters what configurations you use for those services - if they are stateful, how many partitions, how many replicas for partition etc. I recommend you look first at health and figure out whether there are system reports that tell you what may be wrong with your app.

**Comment by TonyGerace** [comment]
Our team really has no need for the application structure of service fabric. So we gave up try to package them all together as it was just not manageable from a developer perspective. We broke them out into different what we call "application domains" (groups of services with some sort of common theme) we now have around 65 services in 10 different applications. I talked to the Service Fabric team at ignite and this was basically how the suggested we do it. We could also package them together into one application during build/release and we might at some point. 

I should have responded to this thread sooner but basically this error resolved when we paired down the amount of services in an application. So it was likely our mismanagement of the Application Manifest.
